src/metpy/calc/basic.py

```diff
@@ -777,15 +778,180 @@ def smooth_gaussian(scalar_grid, n):
+        # applying offsets in needed dimensions
+        data[_trailing_dims(_offset(_pad(n), 0) for n in weights.shape)] = sum(
+            weights[index] * data[_trailing_dims(_offset(_pad(n), index[i] - _pad(n))
+                                                 for i, n in enumerate(weights.shape))]
+            for index in product(*(range(n) for n in weights.shape))
+        )
+
+    return data
+
+
+@exporter.export
+def smooth_rectangular(scalar_grid, size, passes=1):
+    """Filter with a rectangular window smoother.
+
+    Parameters
+    ----------
+    scalar_grid : array-like
+        N-dimensional scalar grid to be smoothed.
+
+    size : int or sequence of ints
+        Shape of rectangle along the trailing dimension(s) of the scalar grid.
+
+    passes : int
+        The number of times to apply the filter to the grid. Defaults to 1.
+
+    Returns
+    -------
+    array-like
+        The filtered scalar grid.
+
+    Notes
+    -----
+    This function can be applied multiple times to create a more smoothed field and will only
+    smooth the interior points, leaving the end points with their original values (this
+    function will leave an unsmoothed edge of size `(n - 1) / 2` for each `n` in `size` around
+    the data). If a masked value or NaN values exists in the array, it will propagate to any
+    point that uses that particular grid point in the smoothing calculation. Applying the
+    smoothing function multiple times will propogate NaNs further throughout the domain.
+
+    See Also
+    --------
+    smooth_window, smooth_circular, smooth_n_point, smooth_gaussian
+
+    """
+    return smooth_window(scalar_grid, np.ones(size), passes=passes)
+
+
+@exporter.export
+def smooth_circular(scalar_grid, radius, passes=1):
+    """Filter with a circular window smoother.
+
+    Parameters
+    ----------
+    scalar_grid : array-like
+        N-dimensional scalar grid to be smoothed. If more than two axes, smoothing is only
+        done along the last two.
+
+    radius : int
+        Radius of the circular smoothing window
+
+    passes : int
+        The number of times to apply the filter to the grid. Defaults to 1.
+
+    Returns
+    -------
+    array-like
+        The filtered scalar grid.
+
+    Notes
+    -----
+    This function can be applied multiple times to create a more smoothed field and will only
+    smooth the interior points, leaving the end points with their original values (this
+    function will leave an unsmoothed edge of size `radius` around the data). If a masked
+    value or NaN values exists in the array, it will propagate to any point that uses that
+    particular grid point in the smoothing calculation. Applying the smoothing function
+    multiple times will propogate NaNs further throughout the domain.
+
+    See Also
+    --------
+    smooth_window, smooth_rectangular, smooth_n_point, smooth_gaussian
+
+    """
+    # Generate the circle
+    size = 2 * radius + 1
```

Comment:
Plus one so it's inclusive in the radius? Maybe note that in the docs.

Reply:
Plus one since the window has to have odd shape in all dimensions. I'll definitely add a note to the docs.